lib/gollum-lib/file.rb

```diff
@@ -131,36 +155,18 @@ def get_disk_reference(name, commit)
       else
         @on_disk_path = pathname.to_path
       end
-      true
+      @on_disk = true
     end
 
-    # Find a file in the given Gollum repo.
-    #
-    # name    - The full String path.
-    # version - The String version ID to find.
-    # try_on_disk - If true, try to return just a reference to a file
-    #               that exists on the disk.
-    #
-    # Returns a Gollum::File or nil if the file could not be found. Note
-    # that if you specify try_on_disk=true, you may or may not get a file
-    # for which on_disk? is actually true.
-    def find(name, version, try_on_disk = false)
-      checked = name.downcase
-      map     = @wiki.tree_map_for(version)
-      commit  = version.is_a?(Gollum::Git::Commit) ? version : @wiki.commit_for(version)
-
-      if (result = map.detect { |entry| entry.path.downcase == checked })
-        @path    = name
-        @version = commit
-
-        if try_on_disk && get_disk_reference(name, commit)
-          @on_disk = true
-        else
-          @blob = result.blob(@wiki.repo)
-        end
-
-        self
-      end
+    def construct_path(name)
+      path = if self.path.include?('/')
```

Comment:
Ternary operator would work here as well:
```ruby
path = self.path.include?('/') ? self.path.sub(/\/[^\/]+$/, '/') : ''
```
Still not great, but shorter. 😉

Reply:
👍